apps/platform_api/src/modules/exchangeKey/exchangeKey.service.ts

```diff
@@ -3,47 +3,89 @@ import { InjectRepository } from '@nestjs/typeorm';
 import { DeepPartial, Repository } from 'typeorm';
 import { ExchangeKey } from './models/exchangeKey.entity';
 import * as ccxt from 'ccxt';
+import { CreateExchangeKeyInput } from './dto/new-exchangeKey.input';
+import {
+  EXCHANGE_KET_INVALID,
+  EXCHANGE_KEY_EXIST,
+  EXCHANGE_KEY_STORE_ERROR,
+  EXCHANGE_NOT_EXIST,
+  SUCCESS,
+} from '@/common/constants/code';
+import { Result } from '@/common/dto/result.type';
+import { UserExchangeService } from '../user-exchange/user-exchange.service';
+import { ExchangeService } from '../exchange/exchange.service';
 
 @Injectable()
 export class ExchangeKeyService {
+  supportedExchanges = ['binance'];
+
   constructor(
     @InjectRepository(ExchangeKey)
-    private ExchangeKeyRepository: Repository<ExchangeKey>
+    private ExchangeKeyRepository: Repository<ExchangeKey>,
+    private UserExchangeRepository: UserExchangeService,
+    private ExchangeRepository: ExchangeService
   ) {}
 
-  // create an exchange key
-  async create(entity: DeepPartial<ExchangeKey>): Promise<boolean> {
-    const verify = this.verifyExchangeKey(entity.exchangeName, entity.accessKey, entity.secretKey);
-    if (!verify) {
-      return false;
+  async createNewExchangeKey(cxt: any, input: CreateExchangeKeyInput): Promise<Result> {
```

Comment:
first up, you got a typo here.

second, I reckon it's better if you just have the function takes in a `userId: string` instead of passing in the entire context.

Reply:
sure!